sky/cli.py

```diff
@@ -4395,6 +4395,83 @@ def _generate_task_with_service(service_yaml_args: List[str],
     return task
 
 
+def _serve_check_service(task: sky.Task):
+
+    spot_use_resources: List[sky.Resources] = [
+        resource for resource in task.resources
+        if resource.use_spot_specified and resource.use_spot
+    ]
+    if len(spot_use_resources) not in [0, len(task.resources)]:
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(
+                'Resources must either all use spot or none use spot. '
+                'To use on-demand and spot instances together, '
+                'use `spot_policy: SpotHedge` and only specify '
+                '`use_spot: true`.')
+
+    assert task.service is not None
+    if task.service.spot_placer is not None:
+        for resource in list(task.resources):
+            if resource.use_spot_specified and not resource.use_spot:
+                logger.info('use_spot will be override to True, '
+                            'because spot placer is enabled.')
+                break
+
+    else:
+        use_spot = False
+        for resource in list(task.resources):
+            if resource.use_spot_specified and resource.use_spot:
+                use_spot = True
+                break
+        if use_spot:
+            yellow = colorama.Fore.YELLOW
+            reset = colorama.Style.RESET_ALL
+            logger.info(f'{yellow}SkyServe uses spot instances as replica '
+                        f'but spot_policy is not specified.{reset} '
+                        f'Consider adding `spot_policy: SpotHedge` to '
+                        'automate spot replica management for better '
+                        'service quality and smaller downtime.')
+
+    if task.service is None:
+        with ux_utils.print_exception_no_traceback():
+            raise RuntimeError('Service section not found.')
+
+    if task.service.spot_placer is not None:
+        if isinstance(task.resources, list):
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError('Does not support both spot_policy '
+                                 'and ordered resources.')
+
+    first_resource_dict = list(task.resources)[0].to_yaml_config()
```

Comment:
assert len(task.resources) >1

Reply:
I think it can be >=1

Reply:
Oh yeah. Thanks for point it out. I mean assert >= 1